bundles/http/Citrix/SAPHana/script.js

```diff
@@ -0,0 +1,913 @@
+        ts_df_ts_overtime_category: timeEntryResponse.d.TimeSheetDataFields.TimeSheetOvertimeCategory,
+        ts_df_ts_task_component: timeEntryResponse.d.TimeSheetDataFields.TimeSheetTaskComponent,
+        ts_df_ts_task_level: timeEntryResponse.d.TimeSheetDataFields.TimeSheetTaskLevel,
+        ts_df_ts_task_type: timeEntryResponse.d.TimeSheetDataFields.TimeSheetTaskType,
+        ts_df_ts_note: timeEntryResponse.d.TimeSheetDataFields.TimeSheetNote,
+        ts_operation: timeEntryResponse.d.TimeSheetOperation,
+        ts_status: timeEntryResponse.d.TimeSheetStatus,
+        user_email: user_email
+    })
+
+    dataStore.save("TimeEntries", timeEntries);
+    dataStore.save("TimeEntriesClock", timeEntries);
+}
+
+async function clockManager({dataStore, client, actionParameters}) {
+    const todayTs = moment().valueOf();
+    let endAt = moment();
+    console.log("## endAt: " + endAt)
+
+    let tokenResponse = await getToken(client);
+    let token = tokenResponse.headers.get("x-csrf-token")
+
+    let cookies = tokenResponse.headers.get("set-cookie").split(",")
+    let cookie = cookies[0].substring(0, cookies[0].indexOf(";")) + ";" +
+        cookies[1].substring(0, cookies[1].indexOf(";"));
+
+    const headersPOST = {
+        'X-CSRF-Token': token,
+        'cookie': cookie
+    }
+
+    const user_email = actionParameters.UserEmail
+    const timeEntryId = actionParameters.TimeEntryId
+    const employeeBusinessPartner =await getEmployeeBusinessPartner(client, user_email)
+    const workAgreement = await getWorkAgreement(client, employeeBusinessPartner)
+
+    let body = {
+        TimeSheetDataFields: {
+            ControllingArea: controllingArea,
+            ReceiverCostCenter: workAgreement.CostCenter,
+            PurchaseOrderItem: actionParameters.PurchaseOrderItem,
+            TimeSheetTaskType: actionParameters.TimeSheetTaskType,
+            TimeSheetTaskLevel: actionParameters.TimeSheetTaskLevel,
+            TimeSheetTaskComponent: actionParameters.TimeSheetTaskComponent,
+            TimeSheetNote: actionParameters.TimeSheetNote,
+            RecordedHours: actionParameters.RecordedHours,
+            RecordedQuantity: actionParameters.RecordedQuantity,
+            HoursUnitOfMeasure: actionParameters.HoursUnitOfMeasure,
+            TimeSheetOvertimeCategory: ""
+        },
+        PersonWorkAgreementExternalID: workAgreement.PersonWorkAgreementExternalID,
+        CompanyCode: workAgreement.CompanyCode,
+        PersonWorkAgreement: workAgreement.PersonWorkAgreement,
+        TimeSheetDate: `/Date(${todayTs})/`,
+        TimeSheetStatus: actionParameters.TimeSheetStatus,
+        TimeSheetOperation: actionParameters.TimeSheetOperation,
+    }
+
+    console.log('## BODY : ' + JSON.stringify(body))
+    console.log("## ActionParameters: " + JSON.stringify(actionParameters))
+    console.log('## TimeEntryId: ' + timeEntryId)
```

Comment:
Do you want to have all the console.logs here? Maybe this was for testing purposes?